src/selectors/layout.test.js

```diff
@@ -1,10 +1,19 @@
-import { mockState } from '../utils/state.mock';
-import { getChartSize, getSidebarWidth, getGraphInput } from './layout';
+import { mockState, prepareState } from '../utils/state.mock';
+import {
+  getChartSize,
+  getSidebarWidth,
+  getGraphInput,
+  getTriggerLargeGraphWarning,
+} from './layout';
 import { updateFontLoaded } from '../actions';
 import reducer from '../reducers';
 import { sidebarWidth } from '../config';
 
 describe('Selectors', () => {
+  describe('getTriggerLargeGraphWarning', () => {
+    expect(getTriggerLargeGraphWarning(mockState.animals)).toEqual(false);
+  });
```

Comment:
Hey this needs more tests - e.g. under what conditions should it return true?

It might be worth using that commit I sent you with the overlarge test dataset creator function.

Reply:
e.g. something like this:
```js
it('returns true when  the dataset is larger than the threshold', () => {
	  // Modify the animals dataset to add so many nodes that it triggers the warning:
	  const data = { ...animals };
	  let extraNodes = [];
	  const iterations = Math.ceil(largeGraphThreshold / data.nodes.length) + 1;
	  new Array(iterations).fill().forEach((d, i) => {
	    const extraNodeGroup = data.nodes.map(node => ({
	      ...node,
	      id: node.id + i
	    }));
	    extraNodes = extraNodes.concat(extraNodeGroup);
	  });
	  data.nodes = data.nodes.concat(extraNodes);
	  const customMockState = prepareState({ data });
	  expect(getTriggerLargeGraphWarning(customMockState)).toEqual(true);
});
```

Reply:
Thanks @richardwestenra - I had actually set up the test this morning but I had not push the test as it is causing the following errors: 
![image](https://user-images.githubusercontent.com/27775658/107984040-ec11f000-6fbe-11eb-9c1a-f2029b7570ad.png)

Clearly from the looks of this, using a large dataset would require further set up work to ensure the nodes are in the right format for the setup, which looks like the random data generator would be the only way to truly generate a dataset fit for this purpose. 

Any suggestions on any quick fix I can do to solve this without a huge amount of setup work?

Reply:
Looks like the issue is that it's passing `null` to the layout graphing algorithm which wasn’t prepared for it. It was doing this in `prepareState` in state.mock.js, on line 27. You can fix it by adding a check for falsey graphState input, e.g. like this:
```js
    // Precalculate graph layout:
    (state) => {
      const layout = state.flags.oldgraph ? graphDagre : graphNew;
      const graphState = getGraphInput(state);
      if (!graphState) {
        return state;
      }
      const graph = layout(graphState);
      return updateGraph(graph);
    },
```